packages/rocketchat-lib/server/functions/saveUser.js

```diff
@@ -130,6 +135,10 @@ RocketChat.saveUser = function(userId, userData) {
 			updateUser.$set.name = userData.name;
 		}
 
+		if (userData.role) {
+			updateUser.$set.roles = [ (userData.role || 'user') ];
+		}
+
```

Comment:
I agree, this would make it simpler and less prone to errors and misunderstandings.